crypto/client.go

```diff
@@ -15,8 +16,16 @@ func New(config Config) (Client, error) {
 		if config.Type != "" {
 			fmt.Println("Invalid encryption type, using default")
 		}
-		return NewAESOpenPGPClient(config)
-	case "aes-openpgp":
-		return NewAESOpenPGPClient(config)
+		fallthrough
+	case ConfigTypeOpenPGP:
+		// Remove other settings
+		// TODO(leon): This is shitty.
+		config.Settings.AESGCMSettings = nil
+		return NewAESOpenPGPClient(config, *config.Settings.AESOpenPGPSettings)
```

Comment:
Why is the `crypto.Config` passed in to the Clients? They should only rely on their specialized configs (e.g. `config.Settings.AESOpenPGPSettings`).

It looks like this is to accommodate the safe's `Load` and `Save` methods, specifically to support upgrades. I wonder if moving the `crypto.Config` to the `Safe` would be a cleaner solution. This would allow us to drop the `Config` method from the `crypto.Client` interface and simplify the implementation of all crypto clients by allowing them to only be concerned with their own, specialized config.